java/src/jmri/jmrit/beantable/BeanTableDataModel.java

```diff
@@ -453,8 +445,14 @@ public void setValueAt(Object value, int row, int col) {
     }
 
     protected void deleteBean(int row, int col) {
-        DeleteBeanWorker worker = new DeleteBeanWorker(getBySystemName(sysNameList.get(row)));
-        worker.execute();
+        jmri.util.ThreadingUtil.runOnGUI(() -> {
+            try {
+                var worker = new DeleteBeanWorker(getBySystemName(sysNameList.get(row)));
+                log.debug("Delete Bean {}", worker.toString());
```

Comment:
No need to change anything, but you don't need `.toString()` when passing something to a logger.  It does those automatically as part of substituting them into the expression if and only if the logging level says the message should be generated.  By passing just the value, not the .toString(), you save a few nanoseconds if the message isn't going to be logged.